### DISCUSSION TITLE
希望可以重构下main.tex还有sty文件，把sty文件重构成为cls文件

### DISCUSSION THREAD

**Discussion by Unknown-Chinese-User** [BYRIO/BUPTBachelorThesis#21] [Q&A]
我是明年会毕业和做毕设吧，今年是有写期末一些报告想着用下模版，<del>然后发现所有论文模版都是book基类，最后试了半天用默认article了</del>，看这个库的时候确实感觉有些费解，主要有以下几个点

- main.tex 按照各个模块的功能分拆成chapter0.tex独立出去到contents文件夹，之后只需要\include{contents/chapter0.tex}就可以了
- 外文原稿和翻译真的是需要的吗，我看其他学校的模版正文结束后就是引用，附录和致谢。然后这个外文模块还用了不少\renewcommand命令，不符合文件和样式分离初衷
- \newpage 命令在一些没必要用的地方使用，就我测试而言\chapter{}就会自带换页，这个时候newpage就没有多大用了。
- 最后想把sty换成cls，就我了解好像只需要修改一些命令即可，不过sty文件里确实有些我没用过的命令，我会多加一些注释

我后面应该会依照我的想法重构一下代码，希望能得到沟通

**Comment by Makiras** [comment]
1. 通过 include 拆分 chapter没有问题。
2. 外文原稿和翻译是需要的，北邮毕设格式要求，请参考教务相关文件（目前已毕业，但要求似乎没变）。
3. 属于代码风格清理，欢迎
4. 欢迎，如果可以整理目录结构更好。

**Comment by Unknown-Chinese-User** [comment]
```
\AtBeginDocument{%
  \renewcommand \thelstlisting
       {\ifnum \c@chapter>\z@ \thechapter-\fi \@arabic\c@lstlisting}%
}
```
在sty文件里面有这个一行，然后main示例文件开头又定义`\def\thelstlisting{}`

这行是取消代码计数器的，我直接把取消代码计数器的替换`\AtBeginDocument`了，是这个意思吗

另外我觉得
```
% 加入目录
\fancypagestyle{plain}{\pagestyle{frontmatter}}
\frontmatter
\tableofcontents % Content
```
这种应该在main.tex中用一行重新定义的语句替代，来让主文件干净一点

**Comment by Unknown-Chinese-User** [comment]
还有怎么提交来审核一下，我暂时只做了把章节分离出去，主文件现在只有56行了
